### PR TITLE
Tenant ensign client configuration

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This adds an Ensign client to Tenant so that Tenant can issue topic requests to Ensign.

Fixes SC-13765

### Type of change

- [ ] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/config/config_test.go [hunk 1]
```diff
@@ -166,6 +174,34 @@ func TestDatabase(t *testing.T) {
...
+	conf := &config.EnsignConfig{
+		Endpoint: "ensign.io:5356",
+		Insecure: true,
+		CertPath: "/path/to/cert",
+		PoolPath: "/path/to/pool",
+	}
+
+	// Endpoint is required
+	conf.Endpoint = ""
+	require.EqualError(t, conf.Validate(), "invalid configuration: ensign endpoint is required", "config should be invalid when bind address is empty")
```
Comment (daniellemaxwell) [inline pkg/tenant/config/config_test.go:187]:
```suggestion
	require.EqualError(t, conf.Validate(), "invalid configuration: ensign endpoint is required", "config should be invalid when endpoint is empty")
```

File: pkg/tenant/config/config.go [hunk 2]
```diff
@@ -58,6 +63,14 @@ type QuarterdeckConfig struct {
 	WaitForReady time.Duration `default:"5m" split_words:"true"`
 }
 
+// Configures the client connection to Ensign.
```
Comment (daniellemaxwell) [inline pkg/tenant/config/config.go:66]: Is `split_words:"true"` required for `Endpoint` and `Insecure`?

**Review by daniellemaxwell — APPROVED** [review]
I made one minor suggestion, but the rest looks good to go. Thanks for walking through this one with me!

File: pkg/tenant/config/config.go (hunk 2 above)
Comment (pdeziel) [reply pkg/tenant/config/config.go:66]: Apparently not, thanks!

**Pull request merged**